File: Linked-List/Doubly-Linked-List/doubly-linked-list/doubly-linked-list-java/remove-doubly-linked-list.java
public void remove(Node currentNode) {
   Node successor = currentNode.next;
   Node predecessor = currentNode.previous;
      
   if (successor != null)
      successor.previous = predecessor;
         
   if (predecessor != null)
      predecessor.next = successor;
         
   if (currentNode == head)
      head = successor;
         
   if (currentNode == tail)
      tail = predecessor;
}

File: Linked-List/Doubly-Linked-List/remove-doubly-linked-list.py
def remove(self, current_node):
    successor_node = current_node.next
    predecessor_node = current_node.prev

    if successor_node is not None:
        successor_node.prev = predecessor_node

    if predecessor_node is not None:
        predecessor_node.next = successor_node

    if current_node is self.head:
        self.head = successor_node

    if current_node is self.tail:
        self.tail = predecessor_node